addons/mail/static/src/models/thread/thread.js

```diff
@@ -724,18 +724,12 @@ function factory(dependencies) {
          * Unsubscribe current user from provided channel.
          */
         async unsubscribe() {
-            if (this.channel_type === 'mail.channel') {
-                return this.async(() => this.env.services.rpc({
-                    model: 'mail.channel',
-                    method: 'action_unfollow',
-                    args: [[this.id]]
-                }));
-            }
-            return this.async(() => this.env.services.rpc({
+            await this.async(() => this.env.services.rpc({
                 model: 'mail.channel',
-                method: 'channel_pin',
-                args: [this.uuid, false]
+                method: 'execute_command',
```

Comment:
This is actually a very interesting problem and one that we have in prod: the impossibility to stay member of a channel but to unpin it.

People who have a lot of channels have a polluted interface (especially messaging menu) which also leads to perf issues due to fetching too much stuff.